README.md

```diff
@@ -25,113 +24,123 @@ Puppet Strings generates HTML documentation for Puppet extensions written in Pup
 
-```
-$ puppet resource package yard provider=gem
-```
+JSON and Markdown output include the reference documentation only. Strings sends JSON output to either STDOUT or to a file. Markdown output is written to a REFERENCE.md file in the module's main directory.
 
-For Puppet Enterprise 3.8 (Linux):
+### Generate documentation in HTML
 
-```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package yard provider=gem
-```
+To generate HTML documentation for a Puppet module, run Strings from that module's directory.
+
+1. Change directory into the module: `cd /modules/<MODULE_NAME>`.
+2. Run the command: `puppet strings`.
 
-For Puppet Enterprise 3.8 (Windows):
+To generate documentation for specific files or directories in a module, run the `puppet strings generate` subcommand and specify the files or directories as a space-separated list. 
 
 ```
-$env:GEM_HOME = "C:\Program Files\Puppet Labs\Puppet Enterprise\sys\ruby\lib\ruby\gems\2.0.0"
-puppet resource package yard provider=gem
+puppet strings generate first.pp second.pp
 ```
 
-### 2. Puppet Enterprise 3.8 in Linux only: Install the redcarpet gem
+To generate documentation for specific directories, run the `puppet strings generate` command and specify the directories:
 
 ```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package redcarpet provider=gem
+$ puppet strings generate 'modules/foo/lib/**/*.rb' 'modules/foo/manifests/**/*.pp' 'modules/foo/functions/**/*.pp' ...
 ```
 
-### 3. Install Puppet Strings
+### Generate documentation in Markdown
 
-Install the [puppet-strings](https://rubygems.org/gems/puppet-strings) gem. To ensure that Strings is installed in the right place, install the gem with Puppet as shown below.
+Strings outputs documentation in Markdown to either an `.md` file or to stdout.
 
-For Puppet 4.x and Puppet Enterprise 2015.2 and later:
+By default, Markdown output generates a `REFERENCE.md` file in the main directory of the module, but you can specify a different location or filename if you prefer. The generated Markdown includes reference information only. The `REFERENCE.md` file is the same format and information we are introducing into Puppet Supported modules.
 
-```
-$ puppet resource package puppet-strings provider=puppet_gem
-```
+1. Change directory into the module: `cd /modules/<MODULE_NAME>`.
+2. Run the command: `puppet strings generate --format markdown`.
 
-For Puppet 3.x:
+   To specify a different file, use the `--out` option and specify the path and filename:
 
-```
-$ puppet resource package puppet-strings provider=gem
-```
+   ```
+   puppet strings generate --format markdown --out docs/INFO.md
+   ```
 
-For Puppet Enterprise 3.8 (Linux):
+### Generate documentation in JSON
 
-```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package puppet-strings provider=gem
-```
+Strings can generate a `.json` file or print JSON to stdout. This can be useful for handling or displaying the data with your own custom applications.
```

Comment:
same with the file format here. Maybe it's ok to say .json and .md. My contention is just that it sounds like you can just say `puppet strings --format json --out blah` and expect to get blah.json, when you won't, you'll get blah.